src/main/java/com/gachigang/ontherun/service/RefreshTokenService.java

```diff
@@ -0,0 +1,77 @@
+package com.gachigang.ontherun.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.gachigang.ontherun.common.enums.TokenType;
+import com.gachigang.ontherun.payload.user.response.AuthenticationResponse;
+import com.gachigang.ontherun.persistence.entity.Token;
+import com.gachigang.ontherun.persistence.entity.User;
+import com.gachigang.ontherun.persistence.repository.TokenRepository;
+import com.gachigang.ontherun.persistence.repository.UserRepository;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpHeaders;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+
+import static com.gachigang.ontherun.common.ApplicationConstants.Security.TOKEN_PREFIX;
+
+@Service
+@RequiredArgsConstructor
+public class RefreshTokenService {
+
+    private final JwtService jwtService;
+    private final UserRepository userRepository;
+    private final TokenRepository tokenRepository;
+
+    public void refreshToken(
```

Comment:
In this method u do a lot of changes in DB, I believe it should be wrapped up with Transaction

Reply:
I will add the Transaction annotation above this method and others in this service.